SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,715 @@
+//
+//  SDLSecondaryTransportManager.m
+//  SmartDeviceLink
+//
+//  Created by Sho Amano on 2018/02/28.
+//  Copyright © 2018 Xevo Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "SDLSecondaryTransportManager.h"
+
+#import "SDLControlFramePayloadConstants.h"
+#import "SDLControlFramePayloadRPCStartServiceAck.h"
+#import "SDLControlFramePayloadTransportEventUpdate.h"
+#import "SDLIAPTransport.h"
+#import "SDLLogMacros.h"
+#import "SDLProtocol.h"
+#import "SDLProtocolHeader.h"
+#import "SDLSecondaryTransportPrimaryProtocolHandler.h"
+#import "SDLStateMachine.h"
+#import "SDLTCPTransport.h"
+#import "SDLTimer.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+// same as SDLProtocol.m
+typedef NSNumber SDLServiceTypeBox;
+
+typedef NS_ENUM(NSInteger, SDLTransportClass) {
+    SDLTransportClassInvalid = 0,
+    SDLTransportClassPrimary = 1,
+    SDLTransportClassSecondary = 2,
+};
+typedef NSNumber SDLTransportClassBox;
+
+typedef NS_ENUM(NSInteger, SDLSecondaryTransportType) {
```

Comment:
These types are also used to describe the primary transport, so perhaps `SDLTransportType` is better?

Reply:
Actually I preferred `SDLTransportType`. The thing was, we already have a `@protocol` named `SDLTransportType`. If I rename the enum to `SDLTransportType` it won't break the build, but it may give confusions to developers. Can I go on to rename the enum?

Some alternatives: `SDLTransportTypeEnum`, `SDLTransportKind`, `SDLTransportUsage`, `SDLTransportEmployment`